compliance_checker/cf/cf_1_6.py

```diff
@@ -156,7 +156,7 @@ def _check_add_offset_scale_factor_type(self, variable, attr_name):
             ) or (  # will short-circuit or if first condition is true
                 isinstance(att, (np.float32, np.float64, float))
                 and variable.dtype
-                in (np.byte, np.short, np.int16, np.int, np.int32, int)
+                in (np.byte, np.short, np.int16, np.int32, np.int64, int)
```

Comment:
I think this can be expressed as `is np.issubdtype(variable.dtype, np.integer)`.

Reply:
`int64` shouldn't actually be here as this is CF 1.6.

Reply:
Also, `issubdtype` on types such as `np.uint8` will return true, while we don't actually want this for CF 1.6.

Reply:
Indeed. I was approaching this from 'catch all ints' point of view, not CF.